pageserver/src/walreceiver.rs

```diff
@@ -64,19 +80,21 @@ pub fn launch_wal_receiver(
             receiver.wal_producer_connstr = wal_producer_connstr.into();
         }
         None => {
-            let receiver = WalReceiverEntry {
-                wal_producer_connstr: wal_producer_connstr.into(),
-            };
-            receivers.insert(timelineid, receiver);
-
-            // Also launch a new thread to handle this connection
-            let _walreceiver_thread = thread::Builder::new()
+            // NOTE: This thread name is checked in the assertion in wait_lsn. If you change
+            // this, make sure you update the assertion too.
+            let wal_receiver_handle = thread::Builder::new()
```

Comment:
We no longer rely on the thread name, since commit 7fab38c5. Looks like this comment got accidentally resurrected while rebasing.